### ISSUE TITLE
Client does not support authentication protocol requested by server. plugin type was = 'caching_sha2_password'

### ISSUE THREAD

**Issue by Fireflasher21** [GRX005/Velocity-PlayTime#7]
As stated in the title, i tried to connect the plugin with default mysql database (mysql  Ver 8.0.40-0ubuntu0.24.04.1 for Linux on x86_64) and a user created with the default authentication method (caching_sha2_password).

All my other plugins work fine but this one throws me this error as soon as i try to connect the plugin to the db.

The user exists, the database exists and the password is correct (confirmed with another working plugin and same creds).

Pls add support.

**Comment by GRX005** [comment]
I see, that's weird, I'll look at it a bit later.

**Comment by LuckyODO** [comment]
I encountered the same issue. The author likely only ensured compatibility with MySQL 5, resulting in the plugin being unusable with MySQL 8 and above.

**Comment by GRX005** [comment]
Yes, it's an issue of the Mariadb dependency being used instead of the Mysql one bc of some past issues, it's been fixed in the latest update since long ago,but I wasn't able to release it yet because of other problems.

**Comment by LuckyODO** [comment]
> Yes, it's an issue of the Mariadb dependency being used instead of the Mysql one bc of some past issues, it's been fixed in the latest update since long ago,but I wasn't able to release it yet because of other problems.

Hey, do you have a rough ETA?

**Comment by LuckyODO** [comment]
Please try #26 

This works for me.
